src/sil/src/SIL_color_publisher_node.cpp

```diff
@@ -2,65 +2,142 @@
 #include "rclcpp/rclcpp.hpp"
 #include "std_msgs/msg/string.hpp"
 
+#include "lifecycle_msgs/msg/transition.hpp"
+#include "rclcpp_lifecycle/lifecycle_node.hpp"
+#include "rclcpp_lifecycle/lifecycle_publisher.hpp"
+
 #include <stdio.h>       
 #include <termios.h>
 #include <unistd.h>
 #include <fcntl.h>
 #include <string>
+#include <chrono>
+#include <thread>
+typedef rclcpp_lifecycle::node_interfaces::LifecycleNodeInterface::CallbackReturn callbackReturn;
 
+//needs to add subscriber coming from the wheels_controller_node.cpp
+//from class Configuration here in Configuration
+class Configuration : public rclcpp_lifecycle::LifecycleNode{
+  public: 
+    explicit Configuration(const std::string& node_name, bool intra_process_comms = false)
+    : rclcpp_lifecycle::LifecycleNode(node_name, 
+      rclcpp::NodeOptions().use_intra_process_comms(intra_process_comms)){}
 
-class Configuration : public rclcpp::Node
-{
-public:
-  Configuration()
-  : Node("sil_node")
-  {
-    
-    this->declare_parameter("sil_path", "/dev/ttyUSB1");
+    void callback(){
```

Comment:
what is the point of this callback?

Reply:
for the timer

Reply:
Why do you want to call this function once per second? The comments say "for demo purposes", I don't think this is actually needed

Reply:
turns out we don't necessarily need a timer